src/k8s/pkg/k8sd/features/cilium/network.go

```diff
@@ -37,22 +37,11 @@ func checkAndSanitizeCiliumVXLAN(port int) error {
 	}
 
 	for _, vxlanDevice := range vxlanDevices {
-		if vxlanDevice.Name == ciliumVXLANDeviceName {
-			// Note(Reza): Currently Cilium tries to bring up the vxlan interface before applying
-			// any configuration changes. If the Cilium vxlan interface has any conflicts with other
-			// interfaces that makes it unable to brought up, Cilium fails to apply configuration
-			// changes. Removing the interface before applying the new manifests is a temporary
-			// workaround. We can remove this block when the following issue gets settled:
-			// https://github.com/cilium/cilium/issues/38581
-			if err := utils.RemoveLink(vxlanDevice.Name); err != nil {
-				return err
-			}
-
-			continue
+		if vxlanDevice.Port == port && vxlanDevice.Name != ciliumVXLANDeviceName {
+			return fmt.Errorf("interface %s uses the same destination port as cilium. Please consider removing that device manually", vxlanDevice.Name)
 		}
-
-		if vxlanDevice.Port == port {
-			return fmt.Errorf("interface %s uses the same destination port as cilium", vxlanDevice.Name)
+		if vxlanDevice.Name == ciliumVXLANDeviceName && vxlanDevice.Port != port {
+			return fmt.Errorf("interface %s uses a different destination port (%d) than the provided config (%d). Please consider removing that device manually", vxlanDevice.Name, vxlanDevice.Port, port)
```

Comment:
This is fine as a check / message however I've bigger questions in terms of the usefulness of this. This check is only performed for the node feature controller is running at. For a multi-node cluster this doesn't work as expected, other nodes might still have the old `cilium_vxlan` interface which will result in Cilium pod failure.

I feel like it is better to point out the port is already in use and for the user to adjust their configuration. We should update our documentation to reflect that this interface removal operation should be performed manually on each node.